javalib/src/main/scala/java/time/YearMonth.scala

```diff
@@ -0,0 +1,242 @@
+
+  def lengthOfMonth(): Int = getMonth().length(isLeapYear())
+
+  def lengthOfYear(): Int = Year.of(year).length()
+
+  def `with`(field: TemporalField, value: Long): YearMonth = {
+    def withYearMonth(y: Int, m: Int): YearMonth = {
+      if (y == year && m == month) this
+      else YearMonth.of(y, m)
+    }
+
+    field match {
+      case YEAR =>
+        withYearMonth(YEAR.checkValidIntValue(value), month)
+
+      case YEAR_OF_ERA =>
+        val isoYear = YEAR.checkValidIntValue(
+          if (year < 1) 1 - value
+          else value
+        )
+        withYearMonth(isoYear, month)
+
+      case MONTH_OF_YEAR =>
+        withYearMonth(year, MONTH_OF_YEAR.checkValidIntValue(value))
+
+      case PROLEPTIC_MONTH =>
+        plusMonths(value - getLong(PROLEPTIC_MONTH))
+
+      case ERA =>
+        requireDateTime(value >= 0 && value <= 1, s"Invalid value for ERA: $value")
+        val era = getLong(ERA)
+        if (value == era) this
+        else withYearMonth(YEAR.checkValidIntValue(1 - year), month)
+
+      case _: ChronoField =>
+        throw new UnsupportedTemporalTypeException("Unsupported field: " + field)
+
+      case _ => field.adjustInto(this, value)
+    }
+  }
+
+  def withYear(year: Int): YearMonth = `with`(YEAR, year)
+
+  def withMonth(month: Int): YearMonth = `with`(MONTH_OF_YEAR, month)
+
+  def plus(amount: Long, unit: TemporalUnit): YearMonth = unit match {
+    case MONTHS    => plusMonths(amount)
+    case YEARS     => plusYears(amount)
+    case DECADES   => plusYears(MathJDK8Bridge.multiplyExact(amount, 10))
+    case CENTURIES => plusYears(MathJDK8Bridge.multiplyExact(amount, 100))
+    case MILLENNIA => plusYears(MathJDK8Bridge.multiplyExact(amount, 1000))
+
+    case ERAS =>
+      val era = getLong(ERA)
+      `with`(ERA, MathJDK8Bridge.addExact(era, amount))
+
+    case _: ChronoUnit =>
+      throw new UnsupportedTemporalTypeException("Unsupported unit: " + unit)
+
+    case _ => unit.addTo(this, amount)
+  }
+
+  def plusYears(years: Long): YearMonth = {
+    if (years == 0) {
+      this
+    } else {
+      // Allowing the Long to overflow to align with JDK implementation
+      val newYear = year + years
+      new YearMonth(YEAR.checkValidIntValue(newYear), month)
+    }
+  }
+
+  def plusMonths(months: Long): YearMonth = {
+    if (months == 0) {
+      this
+    } else {
+      // Allowing the Long to overflow to align with JDK implementation
+      val newProlepticMonth = prolepticMonth + months
+      val newYear = MathJDK8Bridge.floorDiv(newProlepticMonth, 12)
+      val newMonth = MathJDK8Bridge.floorMod(newProlepticMonth, 12) + 1
+      new YearMonth(
+          YEAR.checkValidIntValue(newYear),
+          MONTH_OF_YEAR.checkValidIntValue(newMonth)
+      )
+    }
+  }
+
+  override def minus(amount: Long, unit: TemporalUnit): YearMonth =
+    if (amount != Long.MinValue) plus(-amount, unit)
+    else plus(Long.MaxValue, unit).plus(1, unit)
```

Comment:
it seems that it still needs it but just doing a cast in the return type so it exposes the same type as expected in the Javadoc. Which makes me think that it will surely need the other `minus` impl as well, the one that takes an instance of a `TemporalAmount`. Something like this:

```scala
override def minus(amount: TemporalAmount): YearMonth =
  super.minus(amount).asInstanceOf[YearMonth]
  
override def minus(amount: Long, unit: TemporalUnit): YearMonth =
  super.minus(amount, unit).asInstanceOf[YearMonth]
```